pkg/plugin/test-keeper/plugin/file_pattern_test.go

```diff
@@ -0,0 +1,76 @@
+package plugin_test
+
+import (
+	"fmt"
+
+	"github.com/arquillian/ike-prow-plugins/pkg/plugin/test-keeper/plugin"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("File pattern features", func() {
+
+	var it = func(filePattern, expectedRegex string) {
+		parsed := plugin.ParseFilePatterns([]string{filePattern})
+		Expect(parsed).To(ConsistOf(plugin.FilePattern{Regex: expectedRegex}))
+	}
+
+	Context("File pattern parsing", func() {
+
+		table.DescribeTable(
+			"should parse file patterns to regex",
+			it,
+			should().
+				parsePattern("**/*Test.java").
+				toRegex(`.*/[^/]*Test\.java$`),
```

Comment:
I'm a bit unsure about this syntax anymore, as it contradicts with the actual types. `it`  is, in fact, an assertion. But one can say a similar thing about `matchers_test.go`. Thus let's treat this comment as "Call To Action" to rethink this and make it best of both words - nice readability but also reflecting what is really going on under the hood

Reply:
> it is, in fact, an assertion. But one can say a similar thing about matchers_test.go

Yes, I have the very same feeling when I saw your code in `matchers_test.go`. this is also why I went with this approach.
How about this:
```go
table.DescribeTable(
	"should parse file patterns to regex",
	assertThat,
	pattern("**/*Test.java").isParsedToRegex(`.*/[^/]*Test\.java$`),
	pattern("*/*Test.java").isParsedToRegex(`[^/]*/[^/]*Test\.java$`),
```

Reply:
Yes, this is way better